### PR TITLE
feat: add support for pivot facets query

### PR DESCRIPTION
**Pull request by bietkul**
#### What does this do / why do we need it?
https://www.notion.so/appbase/Pivot-Facets-Hierarchical-dde19c6f5b7b4fc8ae3d7511f83def55

https://www.loom.com/share/36029bf478ba44d1bcbefa96f2ec72ad

### CODE REVIEW

File: plugins/querytranslate/translate_test.go [hunk 1]
```diff
@@ -63,19 +63,6 @@ func TestTranslateQuery(t *testing.T) {
 		_, err := translateQuery(rsQuery, "127.0.0.1")
 		So(err, ShouldBeNil)
 	})
-	Convey("with multiple dataFields for search", t, func() {
```
Comment (siddharthlatest) [inline plugins/querytranslate/translate_test.go]: @bietkul Why remove this test?

**Review by siddharthlatest — CHANGES_REQUESTED** [review]
- [ ] Share the generated query for the component (playground link or cURL is good)
- [ ] Share the generated query for the dependent component (playground link or cURL is good)

- [ ] Code walkthrough

- [ ] Why is multiple data fields in search test removed?

**Comment by bietkul** [comment]
`cURL` request for both `aggs` and `results`
```
curl --location --request POST 'http://foo:bar@localhost:8000/best-buy-dataset/_reactivesearch/validate' \
--header 'Content-Type: application/json' \
--data-raw '{
    "query": [
        {
            "id": "t",
            "dataField": [
                "class.keyword",
                "subclass.keyword"
            ],
            "type": "term",
            "size": 0,
            "aggregationSize": 2,
            "sortBy": "count",
            "value": [
                "DVD SOFTWARE",
                "COMPACT DISC > VINYL"
            ]
        },
        {
            "id": "result",
            "react": {
                "and": "t"
            },
            "size": 5,
            "defaultQuery": {
                "track_total_hits": true
            },
            "includeFields": ["class", "subclass"]
        }
    ]
}'
```

**Review by siddharthlatest — CHANGES_REQUESTED** [review]
- [ ] Edge case: queryFormat value change should change the query
https://www.loom.com/share/86139102401040399c13e5a881a398e2

File: plugins/querytranslate/term_query_test.go [hunk 2]
```diff
@@ -360,3 +360,76 @@ func TestMultiListWithSortByCount(t *testing.T) {
...
+				},
+			},
+		}
+		transformedQuery, err := transformQuery(query)
+		if err != nil {
+			t.Fatalf("Test Failed %v instead\n", err)
+		}
+		convey.So(transformedQuery, convey.ShouldResemble, "{\"preference\":\"Results_127.0.0.1\"}\n{\"_source\":{\"excludes\":[],\"includes\":[\"*\"]},\"query\":{\"bool\":{\"must\":[{\"bool\":{\"must\":{\"bool\":{\"should\":[{\"bool\":{\"must\":[{\"term\":{\"class.keyword\":\"COMPACT DISC\"}},{\"term\":{\"subClass.keyword\":\"VINYL\"}}]}}]}}}}]}}}\n")
+	})
+	convey.Convey("pivot facets: results query with queryFormat as or", t, func() {
```
Comment (siddharthlatest) [inline plugins/querytranslate/term_query_test.go]:
```suggestion
	convey.Convey("pivot facets: results query with queryFormat as and", t, func() {
```

**Review by siddharthlatest — APPROVED** [review]

**Pull request merged**